src/manifolds/Rotations.jl

```diff
@@ -375,12 +375,6 @@ end
 function parallel_transport_direction!(::Rotations{TypeParameter{Tuple{2}}}, Y, p, X, d)
     return copyto!(Y, X)
 end
-function parallel_transport_direction(M::Rotations, p, X, d)
-    expdhalf = exp(d / 2)
-    q = exp(M, p, d)
-    return transpose(q) * p * expdhalf * X * expdhalf
-end
-parallel_transport_direction(::Rotations{TypeParameter{Tuple{2}}}, p, X, d) = X
```

Comment:
Why exactly isn't this needed? IIRC it was a small optimization for static arrays. Is it done on unitary matrix level?

Reply:
These are no longer hit, since we now also allocate first for the vector transports.

Reply:
But we don't if this function gets called directly?

Reply:
If we indeed need them, we would have to add that on layer 2 (the `_vector_transport_` level), but I think the (a) were only fallbacks since these were not unified to allocate first (yet) and the last line is even cheating a bit, since it would be mutating (or returning its input to be precise).

Reply:
> But we don't if this function gets called directly?

They are no longer covered by tests, that is mainly why I removed them. The last update in base moved vector transports also to allocate-first (like we did with retractions).

Reply:
Returning input is not cheating if no change is necessary. And I'd prefer to keep parallel transport non-allocating for `SMatrix` is possible.

Reply:
hm oh that is even PT, that is a bit surprising, I did not rework that, but maybe some fallback is then no longer hitting. hen we maybe should get that back (I was probably not careful enough and thought they were vector transports, those now allocate earlier).

But wer have to come up with tests then, and I am super uncreative with that currently.

Reply:
I will try to add a test then.

Reply:
Most time I spent on finding out about the `mean!` inconsistency and then fixing it. Well – I think this got better now and it is nice to finish this to continue with the other ones that depend on this one.